### PR TITLE
 #572: add the ability to set working directory for run/debug configu…

### PR DESCRIPTION
**Pull request by avesloguzova**
…ration

### CODE REVIEW

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java [hunk 1]
```diff
@@ -86,7 +113,8 @@ public void readExternal(Element element) throws InvalidDataException {
 
   @Nullable
   @Override
-  public final RunningState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment environment) throws ExecutionException {
+  public final RunningState getState(@NotNull Executor executor,
+                                     @NotNull ExecutionEnvironment environment) throws ExecutionException {
```
Comment (deadok22) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: Please avoid spurious formatting changes in your pull requests. It complicates browsing your changes.

File: src/org/intellij/erlang/runconfig/ErlangRunningState.java [hunk 2]
```diff
@@ -105,7 +107,8 @@ public final void setExePath(GeneralCommandLine commandLine) throws ExecutionExc
   }
 
   public final void setWorkDirectory(GeneralCommandLine commandLine) {
-    commandLine.withWorkDirectory(myModule.getProject().getBasePath());
+    String workDirectory = getWorkDirectory();
+    commandLine.withWorkDirectory(workDirectory == null || workDirectory.isEmpty() ? myModule.getProject().getBasePath() : workDirectory);
```
Comment (deadok22) [inline src/org/intellij/erlang/runconfig/ErlangRunningState.java]: Please, use ```com.intellij.openapi.util.text.StringUtil.isEmpty``` to check if a string is null or empty.

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java [hunk 3]
```diff
@@ -41,8 +43,11 @@
 public abstract class ErlangRunConfigurationBase<RunningState extends ErlangRunningState> extends ModuleBasedConfiguration<ErlangModuleBasedConfiguration>
   implements RunConfigurationWithSuppressedDefaultRunAction {
   private ErlangDebugOptions myDebugOptions = new ErlangDebugOptions();
+  private String myWorkDirectory;
 
-  public ErlangRunConfigurationBase(String name, ErlangModuleBasedConfiguration configurationModule, ConfigurationFactory factory) {
+  public ErlangRunConfigurationBase(String name,
+                                    ErlangModuleBasedConfiguration configurationModule,
+                                    ConfigurationFactory factory) {
```
Comment (deadok22) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: Please avoid spurious formatting changes in your pull requests. It complicates browsing your changes.

**Comment by deadok22** [comment]
It's ready to be merged after you fix the problems I've pointed out.

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java [hunk 4]
```diff
@@ -55,6 +58,14 @@ public void setDebugOptions(@NotNull ErlangDebugOptions debugOptions) {
     myDebugOptions = debugOptions;
   }
 
+  public String getWorkDirectory() {
```
Comment (ignatov) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: Please add annotations.

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java [hunk 5]
```diff
@@ -71,6 +82,20 @@ public void writeExternal(Element element) throws WriteExternalException {
   public void checkConfiguration() throws RuntimeConfigurationException {
     ErlangModuleBasedConfiguration configurationModule = getConfigurationModule();
     configurationModule.checkForWarning();
+    checkWorkDirectory();
+  }
+
+  private void checkWorkDirectory() throws RuntimeConfigurationError {
+    if (myWorkDirectory != null && !myWorkDirectory.isEmpty()) {
```
Comment (ignatov) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: StringUtil.isEmpty

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java [hunk 6]
```diff
@@ -71,6 +82,20 @@ public void writeExternal(Element element) throws WriteExternalException {
...
+      File dir = new File(myWorkDirectory);
+      try {
+        if (!dir.isDirectory()) {
+          throw new RuntimeConfigurationError("Incorrect path to working directory.");
+        }
+      } catch (SecurityException e) {
```
Comment (ignatov) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: New line is missing.

File: src/org/intellij/erlang/runconfig/ErlangRunningState.java [hunk 7]
```diff
@@ -87,6 +87,8 @@ public Module getModule() {
 
   public abstract ErlangEntryPoint getEntryPoint() throws ExecutionException;
 
+  public abstract String getWorkDirectory();
```
Comment (ignatov) [inline src/org/intellij/erlang/runconfig/ErlangRunningState.java]: Annotations

File: src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java (hunk 6 above)
Comment (deadok22) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: We don't have ```'catch' on newline``` code style option enabled in this repo. May be we should enable it, I find having having a newline before ```catch``` more convenient, too.

Comment (ignatov) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: Will be cool to have it :)

Comment (deadok22) [inline src/org/intellij/erlang/runconfig/ErlangRunConfigurationBase.java]: https://github.com/ignatov/intellij-erlang/commit/e35d66dd2f3e7f725da9d02ac0cfc144e43cbd19

File: src/org/intellij/erlang/runconfig/ErlangRunningState.java (hunk 7 above)
Comment (avesloguzova) [inline src/org/intellij/erlang/runconfig/ErlangRunningState.java]: Fixed

**Pull request merged**